packages/backend/src/utils/downloader.ts

```diff
@@ -87,6 +87,22 @@ export class Downloader {
     });
   }
 
+  /**
+   * This file takes as argument a location, either a full url or a path
+   * if a path is provided, the base url will be used as origin.
+   * @param location
+   * @protected
+   */
+  protected getRedirect(location: string): string {
+    if (URL.canParse(location)) return location;
+
+    const origin = new URL(this.url).origin;
+    if (URL.canParse(location, origin)) return new URL(location, origin).href;
+
+    console.warn(`malformed location: cannot parse ${location}`);
```

Comment:
I think that if the log trace needs to stay, it needs more context else a user won't understand if you see this trace being printed